zapisy/apps/enrollment/timetable/assets/components/SimpleSummary.vue

```diff
@@ -0,0 +1,140 @@
+<script lang="ts">
+import Vue from "vue";
+import Component from "vue-class-component";
+import { DayOfWeek, nameDay, Group, Course } from "../models";
+
+export class CourseWithGroups extends Course {
+  public groups: Group[];
+  public isOverlapping: Boolean;
+
+  constructor(
+    public course: Course,
+    public courseGroups: Group[],
+    public courseIsOverlapping: Boolean
+  ) {
+    super(course.id, course.name, course.shortName, course.url, course.points);
+    this.groups = courseGroups;
+    this.isOverlapping = courseIsOverlapping;
+  }
+}
+
+const SimpleSummaryProps = Vue.extend({
+  props: {
+    summaryType: {
+      type: String,
+      default: "",
+    },
+    summaryData: {
+      type: Array as () => Array<CourseWithGroups>,
+      default() {
+        return [];
+      },
+    },
+  },
+});
+
+@Component({})
+export default class SimpleSummary extends SimpleSummaryProps {
+  openedCourses: Array<number> = [];
+  openedCategory: Boolean = false;
+
+  public getNameDay(day: DayOfWeek) {
+    return nameDay(day).toLowerCase();
+  }
+
+  public getPoints(data: CourseWithGroups) {
+    return data.isOverlapping ? "–" : data.points;
+  }
+
+  public getTotalPoints(): Number {
+    let sum = 0;
+    this.summaryData.forEach((data) => {
+      sum += data.isOverlapping ? 0 : data.points;
+    });
+    return sum;
+  }
+
+  public toggleCourseView(cid: number) {
+    let index = this.openedCourses.indexOf(cid);
+    if (index > -1) {
+      this.openedCourses.splice(index, 1);
+    } else {
+      this.openedCourses.push(cid);
+    }
+  }
+
+  public isCourseViewOpened(cid: number) {
+    return this.openedCourses.includes(cid);
+  }
+
+  public toggleCategoryView(): Boolean {
+    this.openedCategory = !this.openedCategory;
+    return this.openedCategory;
+  }
+}
+</script>
+
+<template>
+  <tbody>
+    <tr class="table-default" v-if="summaryData.length > 0">
+      <td class="summaryHeader" @click="toggleCategoryView()">
+        <strong>{{ summaryType }}</strong>
+      </td>
+      <td class="summaryHeaderEcts">
+        <div class="table-second-column">
+          {{ getTotalPoints() }}
+        </div>
+      </td>
+    </tr>
+    <template v-for="item in summaryData">
+      <tr v-if="openedCategory" class="table-transparent" :key="item.id">
+        <td @click="toggleCourseView(item.id)" class="table-transparent-data">
+          <a @click="toggleCourseView(item.id)" :href="item.url">
+            {{ item.name }}
+          </a>
+        </td>
+        <td class="table-transparent-data">
+          <div class="table-second-column">
+            {{ getPoints(item) }}
+          </div>
+        </td>
+      </tr>
+      <tr
+        class=""
+        v-if="openedCategory && isCourseViewOpened(item.id)"
+        :key="item.name + item.id"
+      >
```

Comment:
Czy to nie zadziała?
```suggestion
    <template v-if="openedCategory" v-for="item in summaryData">
      <tr class="table-transparent" :key="item.id">
        <td @click="toggleCourseView(item.id)" class="table-transparent-data">
          <a @click="toggleCourseView(item.id)" :href="item.url">
            {{ item.name }}
          </a>
        </td>
        <td class="table-transparent-data">
          <div class="table-second-column">
            {{ getPoints(item) }}
          </div>
        </td>
      </tr>
      <tr
        class=""
        v-if="isCourseViewOpened(item.id)"
        :key="item.name + item.id"
      >
```

Reply:
v-if nie może być w tagu razem z v-for. Wykorzystując dodatkowy wrapper udało mi się osiągnąć to o co chyba Panu chodziło. (https://github.com/iiuni/projektzapisy/pull/1248/commits/0cea03b0a79dbc555f2c76b6c3b303411d2f8810)